.github/workflows/native-tests-daily.yml

```diff
@@ -5,9 +5,12 @@ on:
     # daily at 4:00 UTC
     - cron: "0 4 * * *"
   workflow_dispatch:
+  pull_request:
```

Comment:
This deviates a bit from how other workflows operate. Many operations that are needed in multiple workflows are extracted into `reusable-xxx.yml` that is included from other workflows. Here you could also have the native test part as reusable and include it from here and also from `build-pull-request.yml`. You could use something like https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c82d74dc68ac88c45b434d6d5ce09eeb20c5e900/.github/workflows/build-pull-request.yml#L15 to limit the native tests step to run only when the correct labels are set.